openshift/content/introduction/introduction.adoc

```diff
@@ -2,15 +2,23 @@
 Red Hat Single Sign-On (RH-SSO) is an integrated sign-on solution available as a Red Hat JBoss Middleware for OpenShift containerized image. The {xpaasproduct} image provides an authentication server for users to centrally log in, log out, register, and manage user accounts for web applications, mobile applications, and RESTful web services.
 
 [[sso-templates]]
-Red Hat offers multiple OpenShift application templates utilizing the {xpaasproduct-shortname} image.
+Red Hat offers multiple OpenShift application templates utilizing the {xpaasproduct-shortname} image version number 7.2. These define the resources needed to develop Red Hat Single Sign-On 7.2 server based deployment and can be split into the following two categories:
 
-For RH-SSO 7.2:
+[[passthrough-templates]]
+* Templates using HTTPS and JGroups keystores and a truststore for the RH-SSO server, all prepared beforehand. These secure the TLS communication using link:https://docs.openshift.com/container-platform/latest/architecture/networking/routes.html#passthrough-termination[passthrough TLS termination]:
 
-* *_sso72-https_*: RH-SSO 7.2 backed by internal H2 database on the same pod.
-* *_sso72-mysql_*: RH-SSO 7.2 backed by ephemeral MySQL database on a separate pod.
-* *_sso72-mysql-persistent_*: RH-SSO 7.2 backed by persistent MySQL database on a separate pod.
-* *_sso72-postgresql_*: RH-SSO 7.2 backed by ephemeral PostgreSQL database on a separate pod.
-* *_sso72-postgresql-persistent_*: RH-SSO 7.2 backed by persistent PostgreSQL database on a separate pod.
+** *_sso72-https_*: RH-SSO 7.2 backed by internal H2 database on the same pod.
+** *_sso72-mysql_*: RH-SSO 7.2 backed by ephemeral MySQL database on a separate pod.
+** *_sso72-mysql-persistent_*: RH-SSO 7.2 backed by persistent MySQL database on a separate pod.
+** *_sso72-postgresql_*: RH-SSO 7.2 backed by ephemeral PostgreSQL database on a separate pod.
+** *_sso72-postgresql-persistent_*: RH-SSO 7.2 backed by persistent PostgreSQL database on a separate pod.
+
+[[reencrypt-templates]]
+* Templates using OpenShift's internal link:https://docs.openshift.com/container-platform/latest/dev_guide/secrets.html#service-serving-certificate-secrets[service serving x509 certificate secrets] to automatically create the HTTPS keystore used for serving secure content, the JGroups keystore used for securing JGroups communications, and RH-SSO server truststore. These are securing the TLS communication using link:https://docs.openshift.com/container-platform/latest/architecture/networking/routes.html#re-encryption-termination[re-encryption TLS termination]:
```

Comment:
Generating JGroups kestores doesn't work yet.

Reply:
Works. But uses the deprecated `ENCRYPT` protocol, when `JGROUPS_ENCRYPT_SECRET` variable isn't defined. There's JIRA for ASYM_ENCRYPT support for the future, correct.

Reply:
@Crumby Updated the Introduction section to clarify for x509 templates, the JGroups keystore isn't generated. See last commit.